web-registry/src/hooks/useQueryIfCreditClassCreator.ts

```diff
@@ -0,0 +1,36 @@
+import { useEffect, useState } from 'react';
+
+import { queryParams } from 'lib/ecocredit/api';
+
+import { useWallet } from '../lib/wallet';
+
+export function useQueryIfCreditClassCreator(): boolean {
+  const [isCreator, setCreator] = useState(false);
+  const { wallet } = useWallet();
+  useEffect(() => {
+    const queryIfCreator = async (): Promise<void> => {
+      if (!wallet?.address) {
+        setCreator(false);
+      } else {
+        try {
+          const result = await queryParams();
+          const allowlistEnabled = result.params?.allowlistEnabled;
+          if (allowlistEnabled) {
+            const isCreator =
+              result.params?.allowedClassCreators.includes(wallet.address) ===
+              true;
+            setCreator(isCreator);
+          } else {
+            // if the allowlist is not enabled, anyone can create a credit class
```

Comment:
this is an important setting to be aware of. a given network may not have the allowlist enabled. i.e. redwood does not have the allowlist enabled:

http://redwood.regen.network:1317/regen/ecocredit/v1/params

that means anyone is allowed to create credit classes on the redwood network. whereas in mainnet, the allowlist is enabled:

http://mainnet.regen.network:1317/regen/ecocredit/v1/params

i chose to write this hook so it respects this param.

Reply:
just something to be aware of in terms of testing whether or not this permission is working.